include/mqtt/endpoint.hpp

```diff
@@ -9491,47 +9491,6 @@ class endpoint : public std::enable_shared_from_this<endpoint<Socket, Mutex, Loc
         }
     }
 
-    void send_publish_no_lifetime(
```

Comment:
Sorry to put this in and then take it out, but after considering it more, I don't think that this function needs to have two different versions. The second version essentially only saved a single conditional, after the optimizer finished with it.